### PR TITLE
Avoid possible build dependency on locale and filesystem order

### PR DESCRIPTION
**Pull request by alexey-milovidov**
Changelog category (leave one):
- Build/Testing/Packaging Improvement

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Avoid possible build dependency on locale and filesystem order. This allows reproducible builds.

### CODE REVIEW

**Comment by alexey-milovidov** [comment]
> Stress test (thread) — Fatal message in clickhouse-server.log 

A consequence of OOM:
```
2021.04.25 00:57:47.238894 [ 100001 ] {} <Fatal> BaseDaemon: ########################################
2021.04.25 00:57:47.239193 [ 100001 ] {} <Fatal> BaseDaemon: (version 21.6.1.6674, build id: 540E540865E50E8E0D37EAEA8980C17C5122A914) (from thread 297) (no query) Received signal Segmentation fault (11)
2021.04.25 00:57:47.239408 [ 100001 ] {} <Fatal> BaseDaemon: Address: 0x7f1b07385000 Access: read. Address not mapped to object.
2021.04.25 00:57:47.239646 [ 100001 ] {} <Fatal> BaseDaemon: Stack trace: 0x8c443f0 0x8c1e30d 0x8c1b154 0x8c22295 0x8be5401 0x8c6d5b4 0x8c68acd 0x1636bb2c 0x8c5c61f 0x1638b4a4 0x8c5b177 0x8c5965a 0x7f26703070b3 0x8bc062e
2021.04.25 00:58:00.153272 [ 100001 ] {} <Fatal> BaseDaemon: 5. __sanitizer::GetMemoryProfile(void (*)(unsigned long, unsigned long, bool, unsigned long*, unsigned long), unsigned long*, unsigned long) @ 0x8c443f0 in /usr/bin/clickhouse
2021.04.25 00:58:31.072328 [ 295 ] {} <Fatal> Application: Child process was terminated by signal 11.
```

**Comment by alexey-milovidov** [comment]
@filimonov Please check how does it help for #22113

**Comment by filimonov** [comment]
Manual checks are time-consuming, and the task priority is minor. Doing some fixes and ask smbd to check 'how does it help' = bad way to deal with that task.

File: utils/list-licenses/list-licenses.sh [hunk 1]
```diff
@@ -2,8 +2,9 @@
 
 ROOT_PATH="$(git rev-parse --show-toplevel)"
 LIBS_PATH="${ROOT_PATH}/contrib"
+LC_ALL=C
 
-ls -1 -d ${LIBS_PATH}/*/ | grep -F -v -- '-cmake' | while read LIB; do
+ls -1 -d ${LIBS_PATH}/*/ | grep -F -v -- '-cmake' | sort | while read LIB; do
```
Comment (filimonov) [inline utils/list-licenses/list-licenses.sh:7]: actually ls should already give a result sorted by name.

**Pull request merged**